eclair-core/src/main/scala/fr/acinq/eclair/channel/Channel.scala

```diff
@@ -722,7 +722,7 @@ class Channel(val nodeParams: NodeParams, val wallet: EclairWallet, remoteNodeId
         case Failure(cause) => handleLocalError(cause, d, Some(fee))
       }
 
-    case Event(c@CMD_SIGN, d: DATA_NORMAL) =>
+    case Event(c: Command, d: DATA_NORMAL) if c == CMD_SIGN =>
```

Comment:
The following doesn't work?
```suggestion
    case Event(CMD_SIGN, d: DATA_NORMAL) =>
```

Reply:
That would force us to duplicate `CMD_SIGN` on line 753:

```scala
  case Failure(cause) => handleCommandError(cause, CMD_SIGN)
```

Reply:
True but I think it makes the pattern matching more obvious. And inside this handler you have to know that what you're handling is `CMD_SIGN` so I don't find it too shocking

Reply:
I disagree, duplicating `CMD_SIGN` can only lead ton inconsistency. Let me try to find a better way.

Reply:
I can't find better. Here is my argument for keeping it that way:
- in my proposal, if someone accidentally removes `if c == CMD_SIGN`, then tests will fail because the pattern will match all commands
- in your proposal, if someone accidentally copy-pastes `handleCommandError(cause, CMD_SIGN)`, then compilation and tests will run just fine

Reply:
> if someone accidentally copy-pastes handleCommandError(cause, CMD_SIGN), then compilation and tests will run just fine

That's true, ideally a test should catch that but we can't test everything...

Reply:
https://github.com/scala/bug/issues/11574

Reply:
Awesome find! "variable-binding pattern" is the key word I was looking for. hasn't been added to the release note AFAICT.

Reply:
Done with a7ddc2fd7.